sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGenerateExec.scala

```diff
@@ -604,46 +602,49 @@ abstract class GpuExplodeBase extends GpuUnevaluableUnaryExpression with GpuGene
         indexIntoData = 0
         closeCurrent()
         if (inputIterator.hasNext) {
-          currentBatch = inputIterator.next()
+          spillableCurrentBatch = SpillableColumnarBatch(inputIterator.next(),
+            SpillPriorities.ACTIVE_BATCHING_PRIORITY)
         }
       }
 
       override def hasNext: Boolean = {
-        if (currentBatch == null || indexIntoData >= numArrayColumns) {
+        if (spillableCurrentBatch == null || indexIntoData >= numArrayColumns) {
           fetchNextBatch()
         }
-        currentBatch != null
+        spillableCurrentBatch != null
       }
 
       override def next(): ColumnarBatch = {
-        if (currentBatch == null || indexIntoData >= numArrayColumns) {
-          fetchNextBatch()
-        }
-
         withResource(new NvtxWithMetrics("GpuGenerateExec", NvtxColor.PURPLE, opTime)) { _ =>
-          withResource(new Array[ColumnVector](numExplodeColumns + numOtherColumns)) { result =>
-
-            withResource(GpuProjectExec.project(currentBatch, boundOthersProjectList)) { cb =>
-              (0 until cb.numCols()).foreach { i =>
-                result(i) = cb.column(i).asInstanceOf[GpuColumnVector].getBase.incRefCount()
-              }
-            }
-            if (position) {
-              result(numOtherColumns) = withResource(GpuScalar.from(indexIntoData, IntegerType)) {
-                scalar => ColumnVector.fromScalar(scalar, currentBatch.numRows())
+          // Do projection with boundOthersProjectList and current boundLazyProjectList item
+          val projectCb = GpuProjectExec.projectAndCloseWithRetrySingleBatch(
```

Comment:
Wold it be better to write a `projectWithRetrySingleBatch` that does not close the input?  At least instead of copying the spillable input